error[E0277]: the size for values of type `T` cannot be known at compilation time
 --> $DIR/project_replace_unsized_fn_params.rs:6:8
  |
6 | struct Struct<T: ?Sized> {
  |        ^^^^^^^-^^^^^^^^^
  |        |      |
  |        |      this type parameter needs to be `std::marker::Sized`
  |        doesn't have a size known at compile-time
  |
note: required because it appears within the type `__StructProjectionOwned<T>`
 --> $DIR/project_replace_unsized_fn_params.rs:6:8
  |
6 | struct Struct<T: ?Sized> {
  |        ^^^^^^
  = note: the return type of a function must have a statically known size

error[E0277]: the size for values of type `T` cannot be known at compilation time
 --> $DIR/project_replace_unsized_fn_params.rs:5:1
  |
5 | #[pin_project(project_replace)] //~ ERROR E0277
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ doesn't have a size known at compile-time
6 | struct Struct<T: ?Sized> {
  |               - this type parameter needs to be `std::marker::Sized`
  |
note: required because it appears within the type `Struct<T>`
 --> $DIR/project_replace_unsized_fn_params.rs:6:8
  |
6 | struct Struct<T: ?Sized> {
  |        ^^^^^^
  = note: required by `UnsafeOverwriteGuard`
  = note: this error originates in a derive macro (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the size for values of type `T` cannot be known at compilation time
   --> $DIR/project_replace_unsized_fn_params.rs:7:5
    |
6   | struct Struct<T: ?Sized> {
    |               - this type parameter needs to be `std::marker::Sized`
7   |     f: T,
    |     ^ doesn't have a size known at compile-time

error[E0277]: the size for values of type `T` cannot be known at compilation time
  --> $DIR/project_replace_unsized_fn_params.rs:11:8
   |
11 | struct TupleStruct<T: ?Sized>(T);
   |        ^^^^^^^^^^^^-^^^^^^^^^
   |        |           |
   |        |           this type parameter needs to be `std::marker::Sized`
   |        doesn't have a size known at compile-time
   |
note: required because it appears within the type `__TupleStructProjectionOwned<T>`
  --> $DIR/project_replace_unsized_fn_params.rs:11:8
   |
11 | struct TupleStruct<T: ?Sized>(T);
   |        ^^^^^^^^^^^
   = note: the return type of a function must have a statically known size

error[E0277]: the size for values of type `T` cannot be known at compilation time
  --> $DIR/project_replace_unsized_fn_params.rs:10:1
   |
10 | #[pin_project(project_replace)] //~ ERROR E0277
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ doesn't have a size known at compile-time
11 | struct TupleStruct<T: ?Sized>(T);
   |                    - this type parameter needs to be `std::marker::Sized`
   |
note: required because it appears within the type `TupleStruct<T>`
  --> $DIR/project_replace_unsized_fn_params.rs:11:8
   |
11 | struct TupleStruct<T: ?Sized>(T);
   |        ^^^^^^^^^^^
   = note: required by `UnsafeOverwriteGuard`
   = note: this error originates in a derive macro (in Nightly builds, run with -Z macro-backtrace for more info)
