error: #[pin_project] attribute has been removed
  --> $DIR/remove-attr-from-struct.rs:22:1
   |
22 | #[pin_project] //~ ERROR has been removed
   | ^^^^^^^^^^^^^^
   |
   = note: this error originates in a derive macro (in Nightly builds, run with -Z macro-backtrace for more info)

error: cannot find attribute `pin` in this scope
  --> $DIR/remove-attr-from-struct.rs:18:7
   |
18 |     #[pin] //~ ERROR cannot find attribute `pin` in this scope
   |       ^^^

error: cannot find attribute `pin` in this scope
  --> $DIR/remove-attr-from-struct.rs:11:7
   |
11 |     #[pin] //~ ERROR cannot find attribute `pin` in this scope
   |       ^^^

error[E0277]: `PhantomPinned` cannot be unpinned
  --> $DIR/remove-attr-from-struct.rs:35:5
   |
6  | fn is_unpin<T: Unpin>() {}
   |                ----- required by this bound in `is_unpin`
...
35 |     is_unpin::<A>(); //~ ERROR E0277
   |     ^^^^^^^^^^^^^ within `A`, the trait `Unpin` is not implemented for `PhantomPinned`
   |
   = note: consider using `Box::pin`
note: required because it appears within the type `A`
  --> $DIR/remove-attr-from-struct.rs:10:8
   |
10 | struct A {
   |        ^

error[E0277]: `PhantomPinned` cannot be unpinned
  --> $DIR/remove-attr-from-struct.rs:36:5
   |
6  | fn is_unpin<T: Unpin>() {}
   |                ----- required by this bound in `is_unpin`
...
36 |     is_unpin::<B>(); //~ ERROR E0277
   |     ^^^^^^^^^^^^^ within `B`, the trait `Unpin` is not implemented for `PhantomPinned`
   |
   = note: consider using `Box::pin`
note: required because it appears within the type `B`
  --> $DIR/remove-attr-from-struct.rs:17:8
   |
17 | struct B {
   |        ^

error[E0277]: `PhantomPinned` cannot be unpinned
  --> $DIR/remove-attr-from-struct.rs:40:13
   |
40 |     let _ = Pin::new(&mut x).project(); //~ ERROR E0277,E0599
   |             ^^^^^^^^ within `A`, the trait `Unpin` is not implemented for `PhantomPinned`
   |
   = note: consider using `Box::pin`
note: required because it appears within the type `A`
  --> $DIR/remove-attr-from-struct.rs:10:8
   |
10 | struct A {
   |        ^
   = note: required by `Pin::<P>::new`

error[E0599]: no method named `project` found for struct `Pin<&mut A>` in the current scope
  --> $DIR/remove-attr-from-struct.rs:40:30
   |
40 |     let _ = Pin::new(&mut x).project(); //~ ERROR E0277,E0599
   |                              ^^^^^^^ method not found in `Pin<&mut A>`

error[E0277]: `PhantomPinned` cannot be unpinned
  --> $DIR/remove-attr-from-struct.rs:43:13
   |
43 |     let _ = Pin::new(&mut x).project(); //~ ERROR E0277,E0599
   |             ^^^^^^^^ within `B`, the trait `Unpin` is not implemented for `PhantomPinned`
   |
   = note: consider using `Box::pin`
note: required because it appears within the type `B`
  --> $DIR/remove-attr-from-struct.rs:17:8
   |
17 | struct B {
   |        ^
   = note: required by `Pin::<P>::new`

error[E0599]: no method named `project` found for struct `Pin<&mut B>` in the current scope
  --> $DIR/remove-attr-from-struct.rs:43:30
   |
43 |     let _ = Pin::new(&mut x).project(); //~ ERROR E0277,E0599
   |                              ^^^^^^^ method not found in `Pin<&mut B>`
