error: reference to packed field is unaligned
  --> $DIR/unaligned_references.rs:17:5
   |
17 |     &a.f; //~ ERROR reference to packed field is unaligned
   |     ^^^^
   |
note: the lint level is defined here
  --> $DIR/unaligned_references.rs:1:11
   |
1  | #![forbid(unaligned_references)]
   |           ^^^^^^^^^^^^^^^^^^^^
   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
   = note: for more information, see issue #82523 <https://github.com/rust-lang/rust/issues/82523>
   = note: fields of packed structs are not properly aligned, and creating a misaligned reference is undefined behavior (even if that reference is never dereferenced)

error: reference to packed field is unaligned
  --> $DIR/unaligned_references.rs:18:13
   |
18 |     let _ = &a.f; //~ ERROR reference to packed field is unaligned
   |             ^^^^
   |
   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
   = note: for more information, see issue #82523 <https://github.com/rust-lang/rust/issues/82523>
   = note: fields of packed structs are not properly aligned, and creating a misaligned reference is undefined behavior (even if that reference is never dereferenced)

error: reference to packed field is unaligned
  --> $DIR/unaligned_references.rs:21:5
   |
21 |     &b.f; //~ ERROR reference to packed field is unaligned
   |     ^^^^
   |
   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
   = note: for more information, see issue #82523 <https://github.com/rust-lang/rust/issues/82523>
   = note: fields of packed structs are not properly aligned, and creating a misaligned reference is undefined behavior (even if that reference is never dereferenced)

error: reference to packed field is unaligned
  --> $DIR/unaligned_references.rs:22:13
   |
22 |     let _ = &b.f; //~ ERROR reference to packed field is unaligned
   |             ^^^^
   |
   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
   = note: for more information, see issue #82523 <https://github.com/rust-lang/rust/issues/82523>
   = note: fields of packed structs are not properly aligned, and creating a misaligned reference is undefined behavior (even if that reference is never dereferenced)
